Neuro/Movement/MovementHandler.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Neuro.DependencyInjection;
+using Neuro.Tasks;
+using UnityEngine;
+
+namespace Neuro.Movement;
+
+public class MovementHandler : IMovementHandler
+{
+    public IContextProvider Context { get; set; }
+
+    public Vector2 LastMoveDirection { get; private set; }
+    public Vector2 DirectionToNearestTask { get; private set; }
+
+    public LineRenderer Arrow { get; set; }
+
+    public Vector2? GetForcedMoveDirection(Vector2 actualDirection)
+    {
+        // TODO: This is terrible, MovementHandler should not assign to fields in different handlers
+        ITasksHandler handler = Context.TasksHandler;
+
+        LastMoveDirection = actualDirection;
+        if (handler.CurrentPath.Length > 0 && handler.PathIndex != -1)
```

Comment:
this can be refactored to reduce indentation:
if (<not of this condition>) {
body of the else;
return null;
}

also.. where's the return statement for the main if?

Reply:
I believe it was temporarily removed in main, hence why it is not in the refactoring either,